bool now
---
tuple_ndarray nbv